rust/operator/src/container.rs

```diff
@@ -0,0 +1,778 @@
+                .map(|(name, value)| ContainerPort {
+                    name: Some(name),
+                    container_port: i32::from(value),
+                    protocol: Some("TCP".to_string()),
+                    ..ContainerPort::default()
+                })
+                .collect(),
+            _ => {
+                vec![]
+            }
+        }
+    }
+
+    /// Transform the ProductConfig map structure to a Vector of env vars.
+    fn transform_env_overrides_to_env_vars(
+        env_overrides: Option<&BTreeMap<String, String>>,
+    ) -> Vec<EnvVar> {
+        env_overrides
+            .cloned()
+            .unwrap_or_default()
+            .into_iter()
+            .map(|(k, v)| EnvVar {
+                name: k,
+                value: Some(v),
+                ..EnvVar::default()
+            })
+            .collect()
+    }
+
+    /// Common shared or required container env variables.
+    fn shared_env_vars(hadoop_conf_dir: &str, zk_config_map_name: &str) -> Vec<EnvVar> {
+        vec![
+            EnvVar {
+                name: "HADOOP_HOME".to_string(),
+                value: Some(String::from(Self::HADOOP_HOME)),
+                ..EnvVar::default()
+            },
+            EnvVar {
+                name: "HADOOP_CONF_DIR".to_string(),
+                value: Some(String::from(hadoop_conf_dir)),
+                ..EnvVar::default()
+            },
+            EnvVar {
+                name: "POD_NAME".to_string(),
+                value_from: Some(EnvVarSource {
+                    field_ref: Some(ObjectFieldSelector {
+                        field_path: String::from("metadata.name"),
+                        ..ObjectFieldSelector::default()
+                    }),
+                    ..EnvVarSource::default()
+                }),
+                ..EnvVar::default()
+            },
+            EnvVar {
+                name: "ZOOKEEPER".to_string(),
+                value_from: Some(EnvVarSource {
+                    config_map_key_ref: Some(ConfigMapKeySelector {
+                        name: Some(String::from(zk_config_map_name)),
+                        key: "ZOOKEEPER".to_string(),
+                        ..ConfigMapKeySelector::default()
+                    }),
+                    ..EnvVarSource::default()
+                }),
+                ..EnvVar::default()
+            },
+        ]
+    }
+
+    /// Get the logging command for init containers
+    fn init_container_logging_args(
+        container_log_config: Option<ContainerLogConfig>,
+        container_name: &str,
+    ) -> String {
+        if let Some(ContainerLogConfig {
+            choice: Some(ContainerLogConfigChoice::Automatic(log_config)),
+        }) = container_log_config
+        {
+            product_logging::framework::capture_shell_output(
+                STACKABLE_LOG_DIR,
+                container_name,
+                &log_config,
+            )
+        } else {
+            String::new()
```

Comment:
If an empty string is returned and the arguments are joined with ` && `, doesn't it result in ` &&  && `?

Reply:
Good point https://github.com/stackabletech/hdfs-operator/pull/290/commits/62fd6436a10c6be1a170b3ea1ac30e339f224458